### ISSUE TITLE
File splitting speed

### ISSUE THREAD

**Issue by DnsIs** [rmyorston/busybox-w32#224]
The network has a package of Linux utilities for Windows ([shell.w32-ix86.zip](https://sourceforge.net/projects/win-bash/files/shell-complete/latest/))

I did some tests of the command - **split**

I took file ~ 1Gb and tried (HDD, win2012server):

```
busybox.exe split -b 100m 1GB.file
~30-40sec

split.exe -b 100m 1GB.file
~5-7sec
```

Need to do something.

Thanks.

**Comment by rmyorston** [comment]
Here's what I found:
- This doesn't seem to be an upstream issue. BusyBox ```split``` is faster than coreutils ```split``` on Linux.
- Using a virtual machine to find out what was going on was an exercise in futility. Performance was dominated by the rubbish virtual disks. I had to boot my one-and-only bare-metal Windows machine.
- It appears that Windows isn't happy with the default 1KB buffer ```split``` uses. The bigger the better, though diminishing returns set in after 16KB.

Using a 16KB buffer for Windows speeds up the test case by a factor of about two and doesn't affect the size of the binary.

Prerelease binaries with this patch are in the [usual place](https://frippery.org/files/busybox/prerelease/).

**Comment by DnsIs** [comment]
Yesterday, after writing this post, I thought that since cutting is so slow, I can try cutting the file with `dd`. I was not the only one. Found a [shell script](https://blog.al4.co.nz/2011/03/esxi-and-splitting-a-file-with-dd/) on the internet and converted to [spl.cmd](https://pastebin.com/gAD40jNk). Then I saw your patch, and I realized I had to try again.

I don't know if I'm doing the right thing. But between tests, I decided to empty standby list manually, using the [RaMMap](https://docs.microsoft.com/en-us/sysinternals/downloads/rammap) program.

`RAMMap.exe -Et`

I did each test 3 times. Here's what happened:

```
busybox.exe split -b 100m FILE.bin		~60sec
split.exe -b 100m FILE.bin			~64sec
busybox_pre.exe split -b 100m FILE.bin		~39sec
cpl.cmd FILE.bin				~39sec
```
Strange, `split ` wasn't faster today.

**Ron, thank you. You are very fast and cool.**

PS Sorry for my google-translate-english.

**Comment by rmyorston** [comment]
>I don't know if I'm doing the right thing. But between tests, I decided to empty standby list manually, using the RaMMap program.

I agree, clearing cached file data between tests is the right thing to do. Except I didn't know about RAMMap so I used busybox ```sync```.

>Strange, split wasn't faster today.

Strange, indeed. Did you use RAMMap in your previous tests?

>PS Sorry for my google-translate-english.

No problem, it's perfectly clear.

**Comment by DnsIs** [comment]
> Strange, indeed. Did you use RAMMap in your previous tests?

No. Apparently this played a role.


_It all started with the fact that I needed to parse a 26GB file. I started looking for tools. I like the Lua, but it has problems reading files >4GB, so I'm looking for options. Found another [gsar](http://tjaberg.com/) - a damn fast thing._
